### ISSUE TITLE
WslInstanceName argument completer fails when OutputEncoding is not UTF-8

### ISSUE THREAD

**Issue by erythana** [wikiped/Wsl-IpHandler#25]
Hello,

i installed the module according to the manual and when trying to set it up with the default commands i get the error "Install-WslIpHandler: Cannot validate argument on parameter 'WslInstanceName'. Valid values generator return a null value."

![image](https://user-images.githubusercontent.com/42657063/174592042-61be454b-9fa4-41c1-ae3d-ac4bb9c7ad02.png)
I tried different WslInstanceNames, both "fedoraremix" (which worked with wsl itself) and "fedoraremix (Standard)". When ommiting the parameter the tool asks me as a prompt, but i get the same error message.
I also tried it with quotes and without

Any ideas on this? Thanks

Edit: Pretty sure it has something to do with this output:
![image](https://user-images.githubusercontent.com/42657063/174617089-eba11d76-6487-466b-abbf-a9eacc29ccff.png)
There are blank lines in there, depending on where i put the ConvertFrom-UTF16toUTF8 i get a different error (something encoding related)

**Comment by wikiped** [comment]
Thanks for reporting this.

Would you be able to show output of the following command:

`wls.exe -l`

**Comment by erythana** [comment]
Thanks for your answer,
the output is:
![image](https://user-images.githubusercontent.com/42657063/174766018-da32ff8d-0e4b-4fdd-a8ac-d9b046f9e8f1.png)

**Comment by wikiped** [comment]
And what does this command outputs:

`[Console]::InputEncoding, [Console]::OutputEncoding | select -Exp EncodingName`

**Comment by erythana** [comment]
This is my output:
![image](https://user-images.githubusercontent.com/42657063/174958209-f6305f80-5b28-42d7-b040-2f65be270950.png)

**Comment by wikiped** [comment]
No sure what the problem is exactly.

If I apply the same encodings as you have - everything runs without any errors.

Could you please execute the following command (it is a one-liner) and show the output:

`ipmo Wsl-IpHandler; . (join-path (split-path (gmo Wsl-IpHandler).Path) 'Scripts/Powershell/FunctionsArgumentCompleters.ps1'); Get-WslInstancesNames`

**Comment by erythana** [comment]
I do get NO output from that. Are there absolute paths in the module regarding the user profile?
This computer seems to have the Documents folder in the profile on a separate partition. This Documents folder also contains the power shell modules

**Comment by wikiped** [comment]
The problem seems to be related to the decoding of the output of `wsl.exe -l`.

The module is expecting that output of `wsl.exe -l` is encoded in UTF-16.

And it seems it is not the case here.

What version of WSL are you running? i.e. what does `wsl.exe --status` show?

**Comment by erythana** [comment]
This is the output:
![image](https://user-images.githubusercontent.com/42657063/174981131-e5140c17-c623-4cb7-b739-c7d7f6d56364.png)

**Comment by wikiped** [comment]
Could you please [update](https://github.com/wikiped/Wsl-IpHandler#how-to-update-this-module) the module and then execute the following commands:

```powershell
ipmo Wsl-IpHandler -Force
. (join-path (split-path (gmo Wsl-IpHandler).Path) 'Scripts/Powershell/FunctionsArgumentCompleters.ps1')
wsl.exe -l | Convert-CommandOutput -Convert-CommandOutput -SourceEncoding 'ibm850'
```

**Comment by wikiped** [comment]
To avoid editing the screenshots you might want to run `cd ..` until you reach a level without username.

There is no need to run `Install-WslIpHandler` until output from `Convert-CommandOutput` is exactly two lines (in your case) without blank lines.

To achieve this we need to understand which encoding is used in your terminal as Source/Destination. My initial hypothesis was that we need to convert from `ibm850` (your terminal encoding), But it did not work.

Let's try the following:

`wsl.exe -l | Convert-CommandOutput -DestinationEncoding 'ibm850'`

**Comment by erythana** [comment]
Thanks, but works quite fast in Greenshot so im fine with it.

![image](https://user-images.githubusercontent.com/42657063/175229712-453f281f-6e70-4fe5-9aa4-c93833c5b854.png)

**Comment by wikiped** [comment]
Hopefully the latest commit should fix this issue.

You can check by clicking <kbd>Tab</kbd> after `-WslInstanceName` parameter when entering `Install-WslIpHandler` command.

If the distribution name completes everything else should work as expected, otherwise the issue remains open.

**Comment by erythana** [comment]
Thank you, now everything is working as expected - good job :)
Now i get another error, but try to look at this on my own:
![image](https://user-images.githubusercontent.com/42657063/175306167-4aeccc1c-d5d9-418b-8bdb-9829c73e57cb.png)

**Comment by wikiped** [comment]
Regarding last error: during installation of the linux parts of the module there are some packages being installed that are required to configure ip addresses. Currently installation scripts are configured for Ubuntu and Fedora distributions.

For another distribution to be supported another issue has to be opened and someone has to make PR for this.

**Comment by wikiped** [comment]
Please update the module to see if fix for #26 helps to solve your problem.
